fhir-persistence/src/main/java/com/ibm/fhir/persistence/index/TagParameter.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * (C) Copyright IBM Corp. 2022
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+ 
+package com.ibm.fhir.persistence.index;
+
+
+/**
+ * A token search parameter value
```

Comment:
should probably say "tag" instead of "token".  But I'm not really sure we need this variant...I think its the same structure as TokenParameter...

Reply:
yeah, initially I planned to use specific types along with a visitor to do the application. I may yet refactor a bit to use that approach, but for now we just have separate lists for each type (and then you're correct that tag and token are similar).

Reply:
updated the comment